admin/controllers/administrator.js

```diff
@@ -328,11 +365,48 @@ const saveCheckWindows = async (req, res, next) => {
   return res.redirect('/administrator/check-windows')
 }
 
+/**
+ * Soft delete a check window by id.
+ * @param req
+ * @param res
+ * @param next
+ * @returns {Promise.<*>}
+ */
+const removeCheckWindow = async (req, res, next) => {
+  let checkWindow
+
+  if (!req.params.checkWindowId) {
+    return res.redirect('/administrator/check-windows')
+  }
+
+  try {
+    checkWindow = await checkWindowDataService.fetchCheckWindow(req.params.checkWindowId)
+  } catch (err) {
+    return next(err)
+  }
+
+  if (checkWindow) {
+    if (Date.parse(checkWindow.checkStartDate) < Date.now()) {
+      req.flash('error', 'Deleting an active check window is not allowed.')
+      return res.redirect('/administrator/check-windows')
```

Comment:
It looks to me like this redirect could isn't needed, if the second one is moved outside of the else condition so it always redirects.

Reply:
Done